test/syscalls/linux/msgqueue.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sys/ipc.h>
+#include <sys/msg.h>
+#include <sys/types.h>
+
+#include "test/util/test_util.h"
+
+namespace gvisor {
+namespace testing {
+namespace {
+
+// AutoQueue is a RAII class used to automatically clean message queues.
+class AutoQueue {
+ public:
+  explicit AutoQueue(int id) : id_(id) {}
+
+  ~AutoQueue() {
+    if (id_ >= 0) {
+      EXPECT_THAT(msgctl(id_, IPC_RMID, nullptr), SyscallSucceeds());
+    }
+  }
+
+  int release() {
+    int old = id_;
+    id_ = -1;
+    return old;
+  }
+
+  int get() { return id_; }
+
+ private:
+  int id_ = -1;
+};
+
+TEST(MsgqueueTest, MsgGet) {
+  // Simple creaton and retrieval.
+  AutoQueue queue1(msgget(1, IPC_CREAT));
```

Comment:
Message queues are system-wide resources, so these tests can easily collide with an existing message queue on the host when running native tests.

At a minimum, use [`ftok(3)`](https://man7.org/linux/man-pages/man3/ftok.3.html) to generate a semi-unique ID. See the [shm tests](https://cs.opensource.google/gvisor/gvisor/+/master:test/syscalls/linux/shm.cc;l=142?q=ftok&ss=gvisor%2Fgvisor) for examples.

Reply:
Done